app/services/adjustment_create_service.rb

```diff
@@ -19,17 +19,19 @@ def call
 
     if @adjustment.valid? && enough_inventory_for_decreases?
       ActiveRecord::Base.transaction do
-        # Make the necessary changes in the db
-        @adjustment.save
-        AdjustmentEvent.publish(adjustment)
-        # Split into positive and negative portions.
-        # N.B. -- THIS CHANGES THE ORIGINAL LINE ITEMS ON @adjustment DO **NOT** RESAVE AS THAT WILL CHANGE ANY NEGATIVE LINE ITEMS ON THE ADJUSTMENT TO POSITIVES
-        increasing_adjustment, decreasing_adjustment = @adjustment.split_difference
-        @adjustment.storage_location.increase_inventory(increasing_adjustment.line_item_values)
-        @adjustment.storage_location.decrease_inventory(decreasing_adjustment.line_item_values)
-      rescue InsufficientAllotment, InventoryError => e
-        @adjustment.errors.add(:base, e.message)
-        raise e
+        begin
+          # Make the necessary changes in the db
+          @adjustment.save
+          AdjustmentEvent.publish(adjustment)
+          # Split into positive and negative portions.
+          # N.B. -- THIS CHANGES THE ORIGINAL LINE ITEMS ON @adjustment DO **NOT** RESAVE AS THAT WILL CHANGE ANY NEGATIVE LINE ITEMS ON THE ADJUSTMENT TO POSITIVES
+          increasing_adjustment, decreasing_adjustment = @adjustment.split_difference
+          @adjustment.storage_location.increase_inventory(increasing_adjustment.line_item_values)
+          @adjustment.storage_location.decrease_inventory(decreasing_adjustment.line_item_values)
+        rescue Errors::InsufficientAllotment, InventoryError => e
+          @adjustment.errors.add(:base, e.message)
+          raise ActiveRecord::Rollback
```

Comment:
quick question just out of curiosity. in the PR you write
> When the error was raised, it wouldn't roll back the transaction - so it would still do the adjustment. Again, this has been around for a long time. :(

Raising any error vs an `ActiveRecord::Rollback` wouldn't affect rollbacks happening? Like regardless of what error is raised the rollback should happen, right?

I'm just curious if the changes do address that issue, or if that is not longer an issue, cause I don't really see why either version would have an issue with rolling back.

Reply:
The issue is that the error was being caught, and a value was added to the `errors` of the record. But we then just raised the error again, and at that point (I think?) it was already outside of the transaction, so a rollback didn't happen. I went back and forth a lot with this PR - maybe the issue with the rollback was in one of the intermediate states.